session.go

```diff
@@ -594,6 +594,24 @@ func (s *session) getExecRet(ctx context.Context, sql string) (string, error) {
 	return value, nil
 }
 
+// GetAllSysVars implements GlobalVarAccessor.GetAllSysVars interface.
+func (s *session) GetAllSysVars() (map[string]string, error) {
+	if s.Value(context.Initing) != nil {
+		return nil, nil
+	}
+	sql := `SELECT VARIABLE_NAME, VARIABLE_VALUE FROM %s.%s;`
```

Comment:
Why should we load all global variables from tikv?

Reply:
This makes no difference for performance,
since SysVars hold all the variable names(https://github.com/pingcap/tidb/blob/master/sessionctx/variable/sysvar.go#L99).

Reply:
I care about the performance, not the memory consuming. We only need to load the variables that in the `show variable` statement.